### ISSUE TITLE
[BUG] object '/$LIB/libonion.so' from /etc/ld.so.preload cannot be preloaded

### ISSUE THREAD

**Issue by Bobcat-yang** [eunomia-bpf/eunomia-bpf#346]
I don't know why this bug is triggered.

```
$ sudo ./ecli run https://eunomia-bpf.github.io/eunomia-bpf/sigsnoop/package.json
ERROR: ld.so: object '/$LIB/libonion.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
INFO [faerie::elf] strtab: 0xf1e symtab 0xf58 relocs 0xfa0 sh_offset 0xfa0
INFO [bpf_loader_lib::skeleton::preload::section_loader] User didn't specify custom value for variable filtered_pid, use the default one in ELF
INFO [bpf_loader_lib::skeleton::preload::section_loader] User didn't specify custom value for variable target_signal, use the default one in ELF
INFO [bpf_loader_lib::skeleton::preload::section_loader] load runtime arg (user specified the value through cli, or predefined in the skeleton) for failed_only: Bool(false), real_type=<INT> '_Bool' bits:8 off:0 enc:bool, btf_type=BtfVar { name: "failed_only", type_id: 55, kind: GlobalAlloc }
INFO [bpf_loader_lib::skeleton::preload::section_loader] User didn't specify custom value for variable __eunomia_dummy_event_ptr, use the default one in ELF
INFO [bpf_loader_lib::skeleton] Running ebpf program...
TIME     PID    TPID   SIG    RET    COMM   
21:20:54  1204136 1204134 17  0      grep
21:20:54  1204137 1204134 17  0      grep
21:20:54  1204138 1204134 17  0      wc
21:20:54  1204135 1204134 17  0      ps
21:20:54  1204134 1204133 17  0      prope_sock.sh
21:20:54  1204139 1204133 17  0      date
21:20:54  1204142 1204140 17  0      grep
21:20:54  1204143 1204140 17  0      grep
21:20:54  1204144 1204140 17  0      awk
21:20:54  1204141 1204140 17  0      ps
21:20:54  1204140 1204133 17  0      prope_sock.sh
21:20:54  0     8895   14     0      swapper/8
```

**Comment by Officeyutong** [comment]
We use AppImage to pack executables of eunomia-bpf, shipping a ld.so in the packed AppImage. Seems there is something wrong when the shipped ld.so working with your ld cache

**Comment by yunwei37** [comment]
Current conclusion after tracing the packaging/runtime path:

- this is not an `ecli` attach/load failure
- the program in your report actually ran successfully and emitted events
- the warning is coming from the AppImage startup path before `ecli` itself starts

Reason:
- our Linux release artifact is an AppImage
- the AppImage runner executes a bundled `ld-linux` with a custom `--library-path`
- glibc loader still consults `/etc/ld.so.preload`, which has system-wide effect

So if the host has a preload entry such as `/$LIB/libonion.so`, the loader can print a preload warning during AppImage startup even though `ecli` later runs fine.

In other words, this is a host preload/AppImage loader interaction, not an eBPF runtime bug in `ecli`.

Recommended handling for affected systems:
- if the warning is acceptable, it can be treated as cosmetic
- if you need to avoid it, use a source-built `ecli` instead of the AppImage release binary

I am not planning an application-level fix here, because this happens before the Rust program starts.

**Comment by yunwei37** [comment]
The current packaging/runtime trace confirms this warning happens in the AppImage loader before `ecli` starts, due to the host-wide `/etc/ld.so.preload` entry; the reported eBPF program itself ran and emitted events. There is no application-level fix to make in `ecli`, so closing this as not planned. A source-built `ecli` avoids the AppImage loader path if the warning must be eliminated.
